app/services/targets/create_service.rb

```diff
@@ -11,13 +11,30 @@ def initialize(current_user, params)
     def call
       raise MaxAllowTargetsError unless targets.count < MAX_ALLOWED_TARGETS
 
-      targets.create!(params)
+      Target.transaction do
+        target = targets.create!(params)
```

Comment:
Avoid sending `params` from controllers as much as possible. Services should be clear interfaces of business behavior.